doc/reference/extra-objects.rst

```diff
@@ -0,0 +1,29 @@
+Extra Objects
+-------------
+
+It's possible to specify native object files to be packaged with OCaml
+libraries or linked into OCaml executables. Do this by using the
+``extra_objects`` field of the ``library`` or ``executable`` stanzas.
+For example:
+
+.. code:: dune
+
+    (executable
+     (public_name main)
+     (extra_objects foo bar))
+
+    (rule
+     (targets foo.o bar.o)
+     (deps foo.c bar.c)
+     (action (run ocamlopt %{deps})))
+
+This example builds an executable which is linked against a pair of native
+object files, ``foo.o`` and ``bar.o``. The ``extra_objects`` field takes a
+list of object names, which correspond to the object file names with their path
+and extension omitted.
```

Comment:
Not necessary in this PR but when reading I am wondering why `extra_objects` doesn't use the ending (probably because it uses whatever the native object file extension is) but the `rule` does (because rules don't have any implied filename rules), which makes this a bit strange.